diff --git a/dgl/src/NanoVG.cpp b/dgl/src/NanoVG.cpp
index e464329..1a6e2e8 100644
--- a/dgl/src/NanoVG.cpp
+++ b/dgl/src/NanoVG.cpp
@@ -981,7 +981,7 @@ NanoWidget::NanoWidget(NanoWidget* groupWidget)
       nData(new PrivateData(this))
 {
     pData->needsScaling = true;
-    pData->skipDisplay = true;
+    pData->skipDisplay = false;
     groupWidget->nData->subWidgets.push_back(this);
 }
 
@@ -993,13 +993,8 @@ NanoWidget::~NanoWidget()
 void NanoWidget::onDisplay()
 {
     NanoVG::beginFrame(getWidth(), getHeight());
-    onNanoDisplay();
 
-    for (std::vector<NanoWidget*>::iterator it = nData->subWidgets.begin(); it != nData->subWidgets.end(); ++it)
-    {
-        NanoWidget* const widget(*it);
-        widget->onNanoDisplay();
-    }
+    onNanoDisplay();
 
     NanoVG::endFrame();
 }
